pkg/mcoproducer/mcoproducer.go

```diff
@@ -0,0 +1,146 @@
+)
+
+const replaceKernelModuleScriptTemplate = `#!/bin/bash
+echo "before checking podman images"
+if podman images | grep $IMAGE | grep -q $TAG; then
+    echo "Image $IMAGE:$TAG found in the local registry, removing in-tree kernel module"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -rd /opt $KERNEL_MODULE
+    if [ $? -eq 0 ]; then
+            echo "Succesffully removed the in-tree kernel module $KERNEL_MODULE"
+    else
+            echo "failed to remove in-tree kernel module $KERNEL_MODULE"
+    fi
+    echo "Running container image to insert the oot kernel module $KERNEL_MODULE"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -d /opt $KERNEL_MODULE
+    echo "OOT kernel module $KERNEL_MODULE is inserted"
+else
+   echo "Image $IMAGE:$TAG is not present in local registry, will try after reboot"
+fi
+`
+const pullKernelModuleScriptTemplate = `#!/bin/bash
+if podman image list | grep $IMAGE | grep -q $TAG; then
+    echo "Image $IMAGE:$TAG found in the local registry.Nothing to do"
+else
+    echo "Image $IMAGE:$TAG not found in the local registry, pulling"
+    podman pull --authfile /var/lib/kubelet/config.json $IMAGE:$TAG
+    if [ $? -eq 0 ]; then
+        echo "Image $IMAGE:$TAG has been successfully pulled, rebooting.."
+        reboot
+    else
+        echo "Failed to pull image $IMAGE:$TAG"
+    fi
+fi
+`
+
+const mcoTemplate = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: $MCO_MACHINE_CONFIG_POOL_REF
+  name: $MCO_NAME
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+        - contents: |
+            [Unit]
+            Description=Replace in-tree kernel module with oot kernel module
+            Before=network-pre.target
+            Wants=network-pre.target
+            DefaultDependencies=no
+            [Service]
+            User=root
+            Type=oneshot
+            TimeoutSec=10
+            ExecStartPre=ls /usr/local/bin
+            ExecStart=/usr/local/bin/replace-kernel-module.sh
+            PrivateTmp=yes
+            RemainAfterExit=no
+            TimeoutSec=60
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true
+          name: "replace-kernel-module.service"
+        - contents: |
+            [Unit]
+            Description=Pull oot kernel module image
+            After=network-online.target
+            Wants=network-online.target
+            DefaultDependencies=no
+            [Service]
+            User=root
+            Type=oneshot
+            ExecStart=/usr/local/bin/pull-kernel-module-image.sh
+            PrivateTmp=yes
+            RemainAfterExit=no
+            TimeoutSec=900
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true
+          name: "pull-kernel-module-image.service"
+        - enabled: false
+          mask: true
+          name: crio-wipe.service
```

Comment:
What is this service and why it is needed?

Reply:
this service is removing unused images from the node, that was not pulled by crio/kubelet

Reply:
So this is just a GC for the images that the new service manually pull?